### PR TITLE
fix(reducers): Use Set instead of Map for tracking StickyCells

### LINKED ISSUE #1086: change toggleStickyCell structure to Immutable.Set
need to switch stickyCells structure to an Immutable.Set in `toggleStickyCell` function within `src/notebook/reducers/document.js`

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #1086.

### CODE REVIEW

File: test/renderer/reducers/document-spec.js [hunk 1]
```diff
@@ -180,7 +181,7 @@ describe('toggleStickyCell', () => {
     };
 
     const state = reducers(originalState, action);
-    expect(state.document.getIn(['stickyCells', id])).to.be.undefined;
+    expect(state.document.get('stickyCells').has(id)).to.be.false;
```
Comment (rgbkrk) [inline test/renderer/reducers/document-spec.js]: Can also use [`hasIn` here](https://facebook.github.io/immutable-js/docs/#/Set/hasIn).

File: test/renderer/reducers/document-spec.js [hunk 2]
```diff
@@ -163,12 +164,12 @@ describe('toggleStickyCell', () => {
     }
 
     const state = reducers(originalState, action);
-    expect(state.document.getIn(['stickyCells', id])).to.be.true;
+    expect(state.document.get('stickyCells').has(id)).to.be.true;
```
Comment (rgbkrk) [inline test/renderer/reducers/document-spec.js]: You can use [`hasIn` here](https://facebook.github.io/immutable-js/docs/#/Set/hasIn).

**Review by rgbkrk — APPROVED** [review]
Woo!

**Comment by captainsafia** [comment]
Ah. Good idea. Updated code.

**Comment by rgbkrk** [comment]
Little bit on the commit messages: I'd use the prefix refactor since this isn't fixing a bug, it's refactoring the sticky cell state.

**Comment by captainsafia** [comment]
Bleh. Updated. I keep forgetting refactor is a verb.

**Pull request merged**